src/edu/washington/escience/myria/coordinator/catalog/MasterCatalog.java

```diff
@@ -1790,6 +1790,9 @@ protected Long job(final SQLiteConnection sqliteConnection) throws CatalogExcept
         }
       }).get();
     } catch (InterruptedException | ExecutionException e) {
+      if (e instanceof InterruptedException) {
```

Comment:
Sorry, I was completely wrong about my suggested handling of `InterruptedException` (@jingjingwang straightened me out). It doesn't make sense here to set the interrupted status of the current thread, because the `InterruptedException` was caught on a completely different thread (the thread pool worker where the `Future` was executed). So the correct thing to do here is just what the original code was doing.